docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Machine ID Architecture
+description: How Teleport Machine ID works.
+---
+
+This section provides an overview of Teleport Machine ID's inner workings.
+
+A more in-depth specification of the workings of Machine ID can be found in
+[the Request For Discussion.](https://github.com/gravitational/teleport/blob/master/rfd/0064-bot-for-cert-renewals.md)
+
+## Bot creation
+
+When creating a bot via `tctl`, up to three resources are created:
+
+- Bot user: this will be the user that the Machine ID agent authenticates as.
+- Bot role: the bot user is assigned the bot role, and the bot role contains
+  various permissions that the bot will need to function. For example, the
+  ability to watch the certificate authorities and the ability to 
+  [impersonate roles](#role-impersonation).
+- Token: for [onboarding](#onboarding), a token must exist that allows the
+  Machine ID agent to initially authenticate as the bot user. If an existing
+  token is not specified, then a static single-use token will be created by the
+  Auth Server.
+
+## `tbot`
+
+`tbot` is the binary that acts as the agent for Machine ID.
+
+On initial load, `tbot` uses the [configured join method](#onboarding) to obtain
+a set of credentials for the bot user from the Teleport Auth Service. It can 
+then use these credentials to communicate with the Teleport Auth Service as the 
+bot.
+
+On a configured regular period `tbot` then:
+
+1. Renews these bot user credentials
+2. For each destination provided in `tbot`'s configuration:
+   1. Obtains [role impersonated](#role-impersonation) credentials for the roles
+      specified in the destination configuration from the Auth Service using the
+      bot user credentials.
+   2. Outputs these role impersonated credentials in various formats.
+
+Concurrently to this, `tbot` monitors the Teleport certificate authorities to
+detect certificate rotations. When this occurs, it triggers additional renewals
+to ensure that `tbot` has certificates signed by the newly rotated certificate
+authority.
+
+## Role Impersonation
+
+At the core of Machine ID is the concept of role impersonation.
+
+Role Impersonation allows a user to generate credentials with a set of requested
+roles. The user does not have to hold these roles, but must have been granted 
+permission to impersonate them. The impersonated credentials still include
+the username of the user that generated them, so actions can be attributed to 
+the user.
+
+These credentials can then be used to complete any action that is allowed by the
+role's configured permissions. 
+
+In the case of Machine ID, the bot user is assigned a bot role, which includes 
+permissions to impersonate the roles that the user has configured.
+
+## Onboarding
+
+Onboarding is the process by which `tbot` initially authenticates with the 
+Teleport auth service in order to receive credentials.
+
+Machine ID leverages the existing Token resource within Teleport, with the
```

Comment:
Is this a dynamic configuration resource? Should we say `token`?

Reply:
Changed this to `token`. Do I need to change this to "dynamic configuration resource" ? We seem to just use the term "resource" pretty ubiquitously in documentation already.